### ISSUE TITLE
Should warn same property access on get accessor

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn-analyzers#6087]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Should-warn-same-property-access-on-get-/10097064)._

---
Hi,
I've spent couple hours after a hard to catch typo error and I believe it's a simple catch for VS.

```
private bool hasNewMessage;
public bool HasNewMessage {
get => HasNewMessage; /* property is trying to "get" itself , instead of field.*/
set => SetProperty(ref hasNewMessage, value);
}
```
It causes a crash without a meaningful stack trace, even poor VS crashes if you watch the object with this error.

I've changed my field naming notation to start with underscore from that time on, but it's not a real solution. 
Wouldn't it be perfect if you could point out the problem, as you do very well with other suggestions and warnings.

Regards,
Mehmet

---
#### Feedback Bot on 7/18/2022, 01:22 AM: 

(private comment, text removed)

**Comment by Youssef1313** [comment]
This should probably be part of CA2011

**Comment by Youssef1313** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/1478
